zebrad/tests/acceptance.rs

```diff
@@ -1,30 +1,295 @@
-//! Acceptance test: runs the application as a subprocess and asserts its
+//! Acceptance test: runs zebrad as a subprocess and asserts its
 //! output for given argument combinations matches what is expected.
-//!
-//! Modify and/or delete these as you see fit to test the specific needs of
-//! your application.
-//!
-//! For more information, see:
-//! <https://docs.rs/abscissa_core/latest/abscissa_core/testing/index.html>
-
-#![deny(warnings, missing_docs, trivial_casts, unused_qualifications)]
+
+#![warn(warnings, missing_docs, trivial_casts, unused_qualifications)]
 #![forbid(unsafe_code)]
 
-use abscissa_core::testing::prelude::*;
-use once_cell::sync::Lazy;
+use color_eyre::eyre::Result;
+use std::time::Duration;
+use zebra_test::prelude::*;
+
+// Todo: The following 3 helper functions can probably be abstracted into one
+fn get_child_single_arg(arg: &str) -> Result<zebra_test::command::TestChild> {
+    let (mut cmd, _guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
+
+    cmd.arg(arg)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn2()
+}
+
+fn get_child_multi_args(args: &[&str]) -> Result<zebra_test::command::TestChild> {
+    let (mut cmd, _guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
+
+    cmd.args(args)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn2()
+}
+
+fn get_child_no_args() -> Result<zebra_test::command::TestChild> {
+    let (mut cmd, _guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
+
+    cmd.stdout(Stdio::piped()).stderr(Stdio::piped()).spawn2()
+}
+
+#[test]
+fn generate_no_args() -> Result<()> {
+    zebra_test::init();
+
+    let child = get_child_single_arg("generate");
+    let output = child.unwrap().wait_with_output()?;
+    let output = output.assert_success()?;
+
+    output.stdout_contains(r"# Default configuration values for zebrad.")?;
+
+    Ok(())
+}
+
+#[test]
+fn generate_args() -> Result<()> {
+    zebra_test::init();
+
+    // Invalid free argument
+    let child = get_child_multi_args(&["generate", "argument"]);
+    let output = child.unwrap().wait_with_output()?;
+    let output = output.assert_failure()?;
+
+    output.stdout_contains(r"unexpected free argument `argument`")?;
```

Comment:
This can be too much maintenance if abscissa changes the strings too often. We can maybe just keep the `assert_failure()` without comparing the string in this cases(invalid argument, invalid flag, etc).

Reply:
Removed all these at https://github.com/ZcashFoundation/zebra/pull/690/commits/38207cbe023d972fc9bc83a7deffad7126dadc2c